### DISCUSSION TITLE
Mutex in Spaceflake struct

### DISCUSSION THREAD

**Discussion by gruzovator** [kkrypt0nn/spaceflake#6] [General]
Could you please explain why a mutex is needed in this code:

```
func (w *Worker) GenerateSpaceflakeAt(at time.Time) (*Spaceflake, error) {
       ...
	spaceflake := new(Spaceflake)
	spaceflake.mutex = new(sync.Mutex)
	spaceflake.mutex.Lock()
	w.increment++
	defer spaceflake.mutex.Unlock()
```

This mutex doesn't guard w.increment cause every call of  GenerateSpaceflakeAt creates a new mutex.
Maybe the mutex is not needed here at all if a 'worker' has one goroutine.

Thank you

**Comment by kkrypt0nn** [comment]
Ah yeah right, don't know why I created a Mutex at the same place where I lock it; pretty useless. Mutex should be created when creating a new worker and then using that one I suppose..

**Reply by kkrypt0nn** [reply]
I ended up still editing it and adding it to the `Worker` so that it actually has a use.
